nncf/quantization/algorithms/channel_alignment/algorithm.py

```diff
@@ -0,0 +1,475 @@
+            GraphPattern.METATYPE_ATTR: GraphPattern.NON_PATTERN_NODE_TYPE,
+        }
+        consumer_attrs = {
+            GraphPattern.LABEL_ATTR: "CONV_CONSUMER",
+            GraphPattern.NODE_TYPE_ATTR: self._backend_entity.get_conv_metatypes(),
+        }
+        conv_const_attrs = {
+            GraphPattern.LABEL_ATTR: "CONV_CONSTANT",
+            GraphPattern.METATYPE_ATTR: GraphPattern.NON_PATTERN_NODE_TYPE,
+        }
+
+        use_constant = True
+
+        def get_conv_conv_pattern() -> GraphPattern:
+            conv_conv = GraphPattern()
+            if use_constant:
+                input_node = conv_conv.add_node(**input_attrs)
+                producer_constant = conv_conv.add_node(**conv_const_attrs)
+                consumer_constant = conv_conv.add_node(**conv_const_attrs)
+
+            pattern_conv_producer = conv_conv.add_node(**producer_attrs)
+            pattern_conv_consumer = conv_conv.add_node(**consumer_attrs)
+
+            if use_constant:
+                conv_conv.add_edge(input_node, pattern_conv_producer)
+                conv_conv.add_edge(producer_constant, pattern_conv_producer)
+                conv_conv.add_edge(consumer_constant, pattern_conv_consumer)
+
+            conv_conv.add_edge(pattern_conv_producer, pattern_conv_consumer)
+            return conv_conv
+
+        def get_conv_add_conv_pattern() -> GraphPattern:
+            conv_bias_conv = GraphPattern()
+            if use_constant:
+                input_node = conv_bias_conv.add_node(**input_attrs)
+                producer_constant = conv_bias_conv.add_node(**conv_const_attrs)
+                bias_producer_const = conv_bias_conv.add_node(**bias_const_attrs)
+                consumer_constant = conv_bias_conv.add_node(**conv_const_attrs)
+
+            pattern_conv_producer = conv_bias_conv.add_node(**producer_attrs)
+            pattern_bias_producer = conv_bias_conv.add_node(**bias_attrs)
+            pattern_conv_consumer = conv_bias_conv.add_node(**consumer_attrs)
+
+            if use_constant:
+                conv_bias_conv.add_edge(input_node, pattern_conv_producer)
+                conv_bias_conv.add_edge(producer_constant, pattern_conv_producer)
+                conv_bias_conv.add_edge(consumer_constant, pattern_conv_consumer)
+                conv_bias_conv.add_edge(bias_producer_const, pattern_bias_producer)
+
+            conv_bias_conv.add_edge(pattern_conv_producer, pattern_bias_producer)
+            conv_bias_conv.add_edge(pattern_bias_producer, pattern_conv_consumer)
+            return conv_bias_conv
+
+        pattern = get_conv_conv_pattern()
+        pattern.add_pattern_alternative(get_conv_add_conv_pattern())
+        return pattern
+
+    def _get_node_pairs(self, nncf_graph: NNCFGraph) -> List[Tuple[NNCFNode, Optional[NNCFNode], NNCFNode]]:
+        pairs = []
+        patterns = self._get_target_patterns()
+        for subgraph in nncf_graph.find_matching_subgraphs(patterns):
```

Comment:
You can use `find_matching_subgraphs` for a single pattern in that case you can remove a check on length

Reply:
But then I need to make two identical for loops with add and without add

Reply:
I mean check on length is fragile. Moreover, are you sure that the return order of the node is always the same like:
`conv_in, add_in, conv_out `?

Reply:
But does this function really inserts biases for ONNX?

Reply:
I'm sure because of this sorting https://github.com/openvinotoolkit/nncf/blob/develop/nncf/common/graph/graph_matching.py#L148

Reply:
Ok